src/test/java/test/failedreporter/FailedReporterScenariosTest.java

```diff
@@ -0,0 +1,81 @@
+package test.failedreporter;
+
+import org.testng.Assert;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import org.testng.reporters.FailedReporter;
+import test.BaseTest;
+
+import java.io.File;
+import java.util.UUID;
+
+public class FailedReporterScenariosTest extends BaseTest {
+    private File tempDir = new File(System.getProperty("java.io.tmpdir"));
+
+    @Test
+    public void testFileCreationSkipWhenNoFailuresExist() {
+        File fileLocation = runTests(false);
+        try {
+            Assert.assertFalse(getLocation(fileLocation).exists());
+        } finally {
+            if (fileLocation.exists()) {
+                deleteRecursive(fileLocation);
+            }
+        }
+    }
+
+    @Test
+    public void testFileCreationWhenFailuresExist() {
+        File fileLocation = runTests(true);
+        try {
+            Assert.assertTrue(getLocation(fileLocation).exists());
+        } finally {
+            if (fileLocation.exists()) {
+                deleteRecursive(fileLocation);
+            }
+        }
+    }
+
```

Comment:
I think a test case is missing: having `WithoutFailure` AND `WithFailure` in the suite and check if `TESTNG_FAILED_XML` has only `WithFailure` into it.